game-core/src/main/java/games/strategy/triplea/pbem/AxisAndAlliesForumPoster.java

```diff
@@ -1,241 +1,180 @@
-              }
-              throw new Exception(
-                  "Unknown error, please contact the forum owner and also post a bug to the tripleA development team");
-            }
-          }
-        } else {
-          throw new Exception("Unable to load forum post " + m_topicId);
+  private static void deleteToken(final CloseableHttpClient client, final int userId, final String token)
+      throws IOException {
+    final HttpDelete httpDelete = new HttpDelete(forumUrl + "/api/v2/users/" + userId + "/tokens/" + token);
+    addTokenHeader(httpDelete, token);
+    client.execute(httpDelete).close(); // ignore errors, execute and then close
+  }
+
+  private int getUserId(final CloseableHttpClient client) throws IOException {
+    final JSONObject jsonObject = queryUserInfo(client);
+    checkUser(jsonObject);
+    return jsonObject.getInt("uid");
+  }
+
+  private void checkUser(final JSONObject jsonObject) {
+    if (!jsonObject.has("uid")) {
+      throw new IllegalStateException(String.format("User %s doesn't exist.", getUsername()));
+    }
+    if (jsonObject.getBoolean("banned")) {
+      throw new IllegalStateException("Your account is banned from the forum.");
+    }
+    if (!jsonObject.getBoolean("email:confirmed")) {
+      throw new IllegalStateException("Your email isn't confirmed yet!");
+    }
+  }
+
+  private JSONObject queryUserInfo(final CloseableHttpClient client) throws IOException {
+    final HttpGet post = new HttpGet(forumUrl + "/api/user/" + getUsername());
+    HttpProxy.addProxy(post);
+    try (CloseableHttpResponse response = client.execute(post)) {
+      return new JSONObject(EntityUtils.toString(response.getEntity()));
+    }
+  }
+
+  private NameValuePair newPasswordParameter() {
+    return new BasicNameValuePair("password", getPassword());
+  }
+
+  private String getToken(final CloseableHttpClient client, final int userId) throws IOException {
+    final HttpPost post = new HttpPost(forumUrl + "/api/v2/users/" + userId + "/tokens");
+    post.setEntity(new UrlEncodedFormEntity(
+        Collections.singletonList(newPasswordParameter()),
+        StandardCharsets.UTF_8));
+    HttpProxy.addProxy(post);
+    try (CloseableHttpResponse response = client.execute(post)) {
+      final String rawJson = EntityUtils.toString(response.getEntity());
+      final JSONObject jsonObject = new JSONObject(rawJson);
+      if (jsonObject.has("code")) {
+        final String code = jsonObject.getString("code");
+        if (code.equalsIgnoreCase("ok")) {
+          return jsonObject.getJSONObject("payload").getString("token");
         }
+        throw new IllegalStateException(
+            "Failed to retrieve Token. Code: " + code + " Message: " + jsonObject.getString("message"));
       }
-    } catch (final Exception e) {
-      turnSummaryRef = e.getMessage();
-      return false;
+      throw new IllegalStateException("Failed to retrieve Token, server did not return correct JSON: " + rawJson);
```

Comment:
Just noticed this line is missing [this change](https://github.com/triplea-game/triplea/pull/4352/files#diff-368c8b065896f051c28eabb1a7563e05R150) from #4352.  Again, not sure if it's important enough to warrant another PR.

Reply:
Yeah, I decided to just skip that change.